spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/pulsar/ProducerConfigProperties.java

```diff
@@ -0,0 +1,406 @@
+/*
+ * Copyright 2023-2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.boot.autoconfigure.pulsar;
+
+import java.time.Duration;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+
+import org.apache.pulsar.client.api.CompressionType;
+import org.apache.pulsar.client.api.HashingScheme;
+import org.apache.pulsar.client.api.MessageRoutingMode;
+import org.apache.pulsar.client.api.ProducerAccessMode;
+import org.apache.pulsar.client.api.ProducerCryptoFailureAction;
+
+import org.springframework.boot.context.properties.PropertyMapper;
+import org.springframework.pulsar.core.ProducerBuilderCustomizer;
+import org.springframework.util.unit.DataSize;
+
+/**
+ * Configuration properties used to specify Pulsar producers.
+ *
+ * @author Chris Bono
+ * @since 3.2
+ */
+public class ProducerConfigProperties {
```

Comment:
Could this be nested within `PulsarProperties`?

Reply:
No, it is currently used by the SCSt Pulsar Binder config properties as well https://github.com/spring-cloud/spring-cloud-stream/blob/25af6a76b04598eb31bf2f84e59f45cee742b8ea/binders/pulsar-binder/spring-cloud-stream-binder-pulsar/src/main/java/org/springframework/cloud/stream/binder/pulsar/properties/PulsarBinderConfigurationProperties.java#L33-L46

Reply:
As above. Please nest it and make the appropriate changes in SCSt Pulsar Binder..